### PR TITLE
how to require client/server gems

### PR DESCRIPTION
**Pull request by dancinglightning**

### CODE REVIEW

File: en/docs/lib.md [hunk 1]
```diff
@@ -1,11 +1,45 @@
...
-**Note:** This section is still a work in progress.
+Volt is a client and server framework. Gems exist for both client and server and many gems are not compatible with either client or server environment. This and other reasons (see below) mean that volt does not automatically require all gems at startup.
+
+### Gems for client and server
+
+Use Gemfile and bundler as always. But as mentioned above, you just have to remember to **require** them and eg. ```config/app.rb``` is a good place.
+
+### Client only gems
+
+A growing list of gems exist for use in the browser. Off course opal comes to mind, though that may still be run on any javascript vm.
```
Comment (vais) [inline en/docs/lib.md]: The sentence *Off course opal comes to mind, though that may still be run on any javascript vm.* is probably not necessary here.

File: en/docs/lib.md [hunk 2]
```diff
@@ -1,11 +1,45 @@
...
+
+Opal-browser or opal-jquery are good examples, just like opal-pixi or opal-phantom, which all use the browser or dom api.
```
Comment (vais) [inline en/docs/lib.md]: Opal-browser ~~or~~ *and* opal-jquery...., ~~which all~~ *all of which* use... ...~~dom api~~ *DOM API*

File: en/docs/lib.md [hunk 3]
```diff
@@ -1,11 +1,45 @@
...
+```ruby
+if RUBY_PLATFORM == 'opal'
+  require "browser"
+end
+
+```
+
+### Server only gems
+
+On the other hand some gems may not work on the browser. Opal is still young, and will off course never support c extensions. Also gems used in Tasks should be loaded only on the server as anything loaded into the client takes network bandwidth.
```
Comment (vais) [inline en/docs/lib.md]: ~~on~~ *in* the browser.... ~~off course~~ *of course*

File: en/docs/lib.md [hunk 4]
```diff
@@ -1,11 +1,45 @@
...
+end
+```
+
+### Bundler setup
 
 Gems in the Gemfile are not automatically required in Volt.  This was a decision made for a few reasons.
 
 1. It really really increases boot performance, especially on larger apps. (one of the biggest complaints I hear about rails) Component gems included in dependencies.rb will have the parts in app included by default.
 2. It makes it so unused code doesn't get loaded.
+3. It is not possible to tell which gems are client only.
```
Comment (vais) [inline en/docs/lib.md]: ~~client only~~ *client-only*

**Comment by dancinglightning** [comment]
Actually just remembered that i put my Opal.use_gem("salama")
into the dependencies.rb, and forgot to mention the whole thing.
Along with all those corrections a bit of a mess, sorry.

**Comment by vais** [comment]
@dancinglightning no worries, just nitpicking, you're doing great!
:100: :+1: to you for working on this.

**Comment by dancinglightning** [comment]
preceded by #69

**Pull request closed without merging**